### ISSUE TITLE
wmt jax workload returns ValueError: cannot reshape array of size 15104 into shape (8, newaxis, 256)

### ISSUE THREAD

**Issue by priyakasimbeg** [mlcommons/algorithmic-efficiency#204]
## Description
Traceback: 
```
I1019 05:38:42.785857 140231347427136 workload.py:161] Translating evaluation dataset.                                                                                             
I1019 05:43:30.049049 140231347427136 spec.py:315] Evaluating on the test split.                                                                                                   
I1019 05:43:30.051254 140231347427136 dataset_info.py:358] Load dataset info from /home/kasimbeg/tensorflow_datasets/wmt14_translate/de-en/1.0.0                                   
I1019 05:43:30.052881 140231347427136 dataset_builder.py:351] Reusing dataset wmt14_translate (/home/kasimbeg/tensorflow_datasets/wmt14_translate/de-en/1.0.0)                     
I1019 05:43:30.053009 140231347427136 logging_logger.py:36] Constructing tf.data.Dataset wmt14_translate for split test, from /home/kasimbeg/tensorflow_datasets/wmt14_translate/d$
-en/1.0.0                                                                                                                                                                          
Traceback (most recent call last):                                                                                                                                                 
  File "submission_runner.py", line 498, in <module>                                                                                                                               
    app.run(main)                                                                                                                                                                  
  File "/home/kasimbeg/algorithmic-efficiency/env/lib/python3.7/site-packages/absl/app.py", line 312, in run                                                                       
    _run_main(main, args)                                                                                                                                                          
  File "/home/kasimbeg/algorithmic-efficiency/env/lib/python3.7/site-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "submission_runner.py", line 482, in main
    FLAGS.tokenizer_vocab_path)
  File "submission_runner.py", line 414, in score_submission_on_workload
    tokenizer_vocab_path)
  File "submission_runner.py", line 310, in train_once
    global_step)
  File "/home/kasimbeg/algorithmic-efficiency/algorithmic_efficiency/spec.py", line 324, in eval_model
    global_step=global_step)
  File "/home/kasimbeg/algorithmic-efficiency/algorithmic_efficiency/workloads/wmt/workload.py", line 137, in _eval_model_on_split
    eval_batch = next(self._eval_iters[split])
  File "/home/kasimbeg/algorithmic-efficiency/algorithmic_efficiency/workloads/wmt/workload.py", line 108, in _input_queue_generator
    for batch in iter(ds):
  File "/home/kasimbeg/algorithmic-efficiency/algorithmic_efficiency/data_utils.py", line 29, in shard_numpy_ds
    return jax.tree_map(_prepare, xs)
  File "/home/kasimbeg/algorithmic-efficiency/env/lib/python3.7/site-packages/jax/_src/tree_util.py", line 205, in tree_map
    return treedef.unflatten(f(*xs) for xs in zip(*all_leaves))
  File "/home/kasimbeg/algorithmic-efficiency/env/lib/python3.7/site-packages/jax/_src/tree_util.py", line 205, in <genexpr>
    return treedef.unflatten(f(*xs) for xs in zip(*all_leaves))
  File "/home/kasimbeg/algorithmic-efficiency/algorithmic_efficiency/data_utils.py", line 27, in _prepare
    return x.reshape((local_device_count, -1) + x.shape[1:])
ValueError: cannot reshape array of size 15104 into shape (8,newaxis,256)
```

## Steps to Reproduce
On kasimbeg-5 run:
```
torchrun --redirects 1:0,2:0,3:0,4:0,5:0,6:0,7:0 --standalone --nnodes=1 --nproc_per_node=8 submission_runner.py     --framework=pytorch     --data_dir=~/local_data/imagenet/pytorch     --imagenet_v2_data_dir=~     --experiment_dir=~     --experiment_name=target_setting     --workload=imagenet_resnet     --submission_path=target_setting_runs/pytorch_nesterov.py     --tuning_search_space=target_setting_runs/imagenet_resnet/tuning_search_space.json
*An ordered list of steps to recreating the issue*
```

## Source or Possible Fix
It looks like the data somehow does not have dimension 256. The batch_size in get_batch_size is ok, and 15104 is divisible by 8, but the result is not divisible by 256.

**Comment by runame** [comment]
Yes, this is a known issue, see my comment in PR #190 (the remainder batch size of the test split % N_GPUs != 0). If you want to run it anyway, you can just set `num_test_examples` to 3000 and slice the split string in `_build_input_pipeline` like [here](https://github.com/mlcommons/algorithmic-efficiency/blob/main/algorithmic_efficiency/workloads/wmt/workload.py#L94-L97) (without the +1).

**Comment by runame** [comment]
Will be addressed by #119.

**Comment by priyakasimbeg** [comment]
Ah ok nice, thanks Runa!
